tests/nd_item/nd_item_mem_fence.cpp

```diff
@@ -16,8 +16,32 @@ using namespace sycl_cts;
 class mem_fence_kernel;
 void test_mem_fence(util::logger &log, cl::sycl::queue &queue) {
   /* set workspace size */
-  const int globalSize = 64;
-  const int localSize = 2;
+  constexpr size_t globalSize = 64;
+  size_t localSize = 2;
+
+  /* adjust work-group size */
+  {
+    /* check work-group size device limit */
+    auto device = queue.get_device();
+    auto maxDeviceWorkGroupSize =
+        device.template get_info<cl::sycl::info::device::max_work_group_size>();
+    localSize = (maxDeviceWorkGroupSize < localSize) ? maxDeviceWorkGroupSize
+                                                     : localSize;
+
+    /* check work-group size kernel limit */
+    /* this check requires online compiler */
+    auto devices = queue.get_context().get_devices();
+    if (!is_compiler_available(devices) || !is_linker_available(devices))
+      return;
```

Comment:
If the devices being targeted do not support online compilation this function returns so the test is skipped, is this the intention?

Reply:
Yes. Online compilation is required to query `cl::sycl::info::kernel_work_group::work_group_size` property. We must make sure that value for this property is >= 2 to test `mem_fence` method.

Reply:
This is a good motivation showing why we need more comments in the code, because otherwise there are more questions during review... Feel free to add this as a comment for clarification. :-)

Reply:
Added more details in the comments.
@AerialMantis, please, let me know if any changes are required.

Reply:
Great clarification!
Thanks.